core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -492,4 +544,27 @@ private void maybeInitializeKeepAliveHandler(ChannelHandlerContext ctx) {
             keepAliveHandler.initialize(ctx);
         }
     }
+
+    private class AggregatingRequestTimeoutTask implements CancellationTask {
+
+        private final ChannelHandlerContext ctx;
+        private final DecodedHttpRequest req;
+
+        AggregatingRequestTimeoutTask(ChannelHandlerContext ctx, DecodedHttpRequest req) {
+            this.ctx = ctx;
+            this.req = req;
+        }
+
+        @Override
+        public boolean canSchedule() {
+            return req.isOpen();
+        }
+
+        @Override
+        public void run(Throwable cause) {
+            assert !req.isFired();
+            discarding = true;
```

Comment:
> This is a change in behavior, so I want to make sure that this is intentional. 

The request is timed out because the client didn't send the body yet, so I thought we should close the connection. We cannot reuse it.